src/cljs/rems/administration/workflow.cljs

```diff
@@ -0,0 +1,138 @@
+(ns rems.administration.workflow
+  (:require [clojure.string :as str]
+            [re-frame.core :as rf]
+            [rems.administration.administration :refer [administration-navigator-container]]
+            [rems.administration.components :refer [inline-info-field]]
+            [rems.atoms :refer [external-link info-field readonly-checkbox]]
+            [rems.collapsible :as collapsible]
+            [rems.common-util :refer [andstr]]
+            [rems.spinner :as spinner]
+            [rems.text :refer [localize-time text text-format]]
+            [rems.util :refer [dispatch! fetch put!]]))
+
+(rf/reg-event-fx
+ ::enter-page
+ (fn [{:keys [db]} [_ workflow-id]]
+   {:db (assoc db ::loading? true)
+    ::fetch-workflow [workflow-id]}))
+
+(defn- fetch-workflow [workflow-id]
+  (fetch (str "/api/workflows/" workflow-id)
+         {:handler #(rf/dispatch [::fetch-workflow-result %])}))
+
+(rf/reg-fx ::fetch-workflow (fn [[workflow-id]] (fetch-workflow workflow-id)))
+
+(rf/reg-event-db
+ ::fetch-workflow-result
+ (fn [db [_ workflow]]
+   (-> db
+       (assoc ::workflow workflow)
+       (dissoc ::loading?))))
+
+(rf/reg-sub ::workflow (fn [db _] (::workflow db)))
+(rf/reg-sub ::loading? (fn [db _] (::loading? db)))
+
+(defn- back-button []
+  [:button.btn.btn-secondary
+   {:on-click #(dispatch! "/#/administration/workflows")}
+   (text :t.administration/back)])
+
+(defn- to-create-workflow []
+  [:a.btn.btn-primary
+   {:href "/#/administration/create-workflow"}
+   (text :t.administration/create-workflow)])
+
+(defn get-localized-value [field key language]
+  (key (first (filter (comp #{(name language)} :langcode)
+                      (:localizations field)))))
+
+(defn license-view [license language]
+  (into [:div.form-item
+         [:h4 (text-format :t.administration/license-field (get-localized-value license :title language))]
+         [inline-info-field (text :t.administration/title) (:title license)]]
+        (concat (for [localization (:localizations license)]
+                  [inline-info-field (str (text :t.administration/title)
+                                          " "
+                                          (str/upper-case (:langcode localization))) (:title localization)])
+                [[inline-info-field (text :t.administration/type) (:type license)]
+                 (when (:textcontent license)
+                   (case (:type license)
+                     "link" [inline-info-field (text :t.create-license/external-link) [:a {:target :_blank :href (:textcontent license)} (:textcontent license) " " [external-link]]]
+                     "text" [inline-info-field (text :t.create-license/license-text) (:textcontent license)]
+                     nil))]
+                (when (= "link" (:type license))
```

Comment:
I don't get it why there is twice check if license type is link?

Reply:
I thought it's more readable this day when there is one check per field.  I.e. there is a field for the type (that must do the typecase) and there is another field that is only shown for link licenses. Therefore two checks.